ompi/mpi/java/c/Makefile.am

```diff
@@ -44,7 +46,7 @@ libmpi_java_la_SOURCES = \
 	mpi_Status.c \
 	mpi_Win.c
 
-libmpi_java_la_LIBADD = $(top_builddir)/ompi/libmpi.la
+libmpi_java_la_LIBADD = -ldl $(top_builddir)/ompi/libmpi.la
```

Comment:
@ggouaillardet Ah, right -- duh!  So do we already have protection for only building this Java stuff when we know we can build it as a shared library?  (Forgive me -- it's been eons since I've looked in this part of the code...)  Also, do we know that the library is *always* `-ldl`?  Can it be called other things (e.g., on Cygwin)?